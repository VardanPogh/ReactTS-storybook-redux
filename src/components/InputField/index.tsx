import React, { ReactComponentElement } from 'react'
import TextField from '@material-ui/core/TextField'
import { Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

interface Props {
  title: string
  type?: string
  icon?: ReactComponentElement<any>
  variant?: string
  id?: string
  onChange?(e: any): void
  defaultValue?: string
}

const useStyles = makeStyles((theme) => ({
  margin: {
    margin: theme.spacing(1),
  },
}))

const InputField: React.FC<Props> = ({
  title,
  type,
  variant,
  icon,
  id,
  onChange,
  defaultValue,
}) => {
  const classes = useStyles()
  switch (variant) {
    case 'withIcon':
      return (
        <div className={classes.margin}>
          <Grid container spacing={1} alignItems="flex-end">
            <Grid item>{icon}</Grid>
            <Grid item>
              <TextField
                id={id}
                label={title}
                type={type}
                onChange={onChange}
                defaultValue={defaultValue}
              />
            </Grid>
          </Grid>
        </div>
      )
    default:
      return (
        <TextField
          id={id}
          label={title}
          variant="outlined"
          type={type}
          onChange={onChange}
          defaultValue={defaultValue}
        />
      )
  }
}

export default InputField
