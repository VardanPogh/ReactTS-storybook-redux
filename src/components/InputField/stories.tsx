import React from 'react'
import { withKnobs, text, select } from '@storybook/addon-knobs'
import PersonIcon from '@material-ui/icons/Person'
import InputField from '.'

export default {
  title: 'InputField',
  component: InputField,
  decorators: [withKnobs],
}
const typeOptions = {
  Text: 'text',
  Password: 'password',
  Number: 'number',
}

const variantOptions = {
  Outlined: 'outlined',
  WithIcon: 'withIcon',
}

export const Default: React.FC = () => (
  <InputField
    title={text('Title', 'Login')}
    type={select('Type', typeOptions, typeOptions.Text)}
    variant={select('Variant', variantOptions, variantOptions.Outlined)}
    icon={<PersonIcon />}
  />
)
