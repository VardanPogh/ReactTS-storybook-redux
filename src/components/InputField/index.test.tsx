import React from 'react'
import { render } from '@testing-library/react'

import InputField from '.'

describe('inputField', () => {
  test('Loads inputField', () => {
    const { container } = render(
      <InputField title="Login" type="text" variant="outlined" />
    )
    expect(container).toMatchSnapshot()
  })
})
