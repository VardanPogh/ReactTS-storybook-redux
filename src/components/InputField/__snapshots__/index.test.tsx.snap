// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputField Loads inputField 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      Login
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-4"
        >
          <span>
            Login
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
